engines/macventure/macventure.cpp

```diff
@@ -958,9 +960,19 @@ Common::String MacVentureEngine::getCommandsPausedString() const {
 
 Common::Path MacVentureEngine::getFilePath(FilePathID id) const {
 	if (id <= 3) { // We don't want a file in the subdirectory
-		return Common::Path(_filenames->getString(id));
+
+		// Convert from MacRoman encoding to unicode for punycode encoding
+		Common::String result = Common::convertToU32String(_filenames->getString(id).c_str(), Common::kMacRoman);
+
+		return Common::Path(Common::punycode_encode(result));
 	} else { // We want a game file
-		return Common::Path(_filenames->getString(3) + "/" + _filenames->getString(id));
+
+		// Convert from MacRoman encoding to unicode for punycode encoding
+		Common::String part1 = Common::convertToU32String(_filenames->getString(3).c_str(), Common::kMacRoman);
```

Comment:
I do not understand this part: the punycoded name IS in MacRoman. E.g. punycode is a way of encoding non-ASCII lines, it does not look into encoding itself...

Reply:
When the game requests a file with "Déjà vu" in it, the "é" is encoded as 0x8E (consistent with MacRoman) whereas in the dumped files "é" is punycoded based on its unicode representation, i.e 0xE9.

So, I convert from MacRoman and then punycode it to match the file name on disk.